muswarmadmin/repositories.py

```diff
@@ -2,14 +2,47 @@
 import os
 from aiosparql.syntax import IRI, Literal
 from shutil import rmtree
-
 import muswarmadmin.pipelines
 from muswarmadmin.prefixes import Doap, SwarmUI
 
 
 logger = logging.getLogger(__name__)
 
 
+async def repository_has_location(app, pipeline):
+    """
+    Check if the repository associated with the pipeline
+    has a location (git url).
+    """
+    result = await app.sparql.query("""
+        ASK
+        FROM {{graph}}
+        WHERE {
+         ?repository ?p {{pipeline}} .
+         ?repository doap:location ?location .
+        }
+    """, pipeline=pipeline)
+    return result['boolean']
+
+
+async def get_repository_drc(app, pipeline):
+    """
+    Get DockerCompose file associated with a given Pipeline.
+    """
+    result = await app.sparql.query("""
+        SELECT DISTINCT ?drctext
+        WHERE {
+         ?repository ?p {{pipeline}} .
```

Comment:
`?p` sounds like any predicate will be good. This isn't the case, the predicate is `swarmui:pipelines` or something